### PR TITLE
Fix: add TypeScript support to eslint

### LINKED ISSUE #289: ESLint does not lint TypeScript files
The actual configuration of ESLint supports only JavaScript files (`js` and `jsx`), but this is a TypeScript project.
We need to add linting support for `ts` and `tsx` files.

### PR DESCRIPTION
**Pull request by andreaforni**
Closes #289 

ESLint was not linting TypeScript files, only JavaScript ones.

This PR:
- Adds TypeScript linting support.
- Adds new script: `lint:fix` to automatically fix linting problems.
- Updates the lint-staged command to lint TypeScript files too.
- Fixes all the lint issues (for this reason, you see a lot of changes).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
   "scripts": {
     "build": "tsup",
     "lint": "eslint .",
+    "lint:fix": "eslint . --fix",
```
Comment (simoneb) [inline package.json:13]: last I remember, file extensions had to be provided as cli flags to extend the defaults, using the --ext (IIRC) flag

Comment (andreaforni) [reply package.json:13]: Hi Simone, you are right, but based on my tests, ESLint works correctly on both JS and TS files without setting `--ext`. This is explained here: https://github.com/typescript-eslint/typescript-eslint/issues/3824

In summary:
The `--ext` default value is "_`.js` and the files that match the `overrides` entries of your configuration_" ([eslint doc](https://eslint.org/docs/latest/use/command-line-interface#--ext)) and the `@typescript-eslint/recommended` adds the `*.ts` and `*.tsx` extensions to the `overrides` entries.

Comment (simoneb) [reply package.json:13]: ha, makes sense, thanks 👌

**Review by marceloFerreira90 — APPROVED** [review]
LGTM

**Pull request merged**